source/read-write-pref.txt

```diff
@@ -0,0 +1,261 @@
+methods to the ``getDatabase()`` method. The code configures the following
+settings:
+
+- ``primaryPreferred`` read preference: Read operations retrieve data from 
+  the primary replica set member, or secondary members if the primary is unavailable
+- ``AVAILABLE`` read concern: Read operations return the instance's most recent data 
+  without guaranteeing that the data has been written to a majority of the replica
+  set members
+- ``MAJORITY`` write concern: The majority of all replica set members
+  must acknowledge the write operation
+
+.. literalinclude:: /includes/read-write-pref.scala
+    :language: scala
+    :dedent:
+    :start-after: start-database-settings
+    :end-before: end-database-settings
+
+.. _scala-read-write-collection:
+
+Collection Configuration
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+This example shows how to set the read preference, read concern, and
+write concern of a collection called ``test_collection`` by chaining their setter
+methods to the ``getCollection()`` method. The code configures the following
+settings:
+
+- ``secondaryPreferred`` read preference: Read operations retrieve data from 
+  secondary replica set members, or the primary members if no secondaries are available
+- ``AVAILABLE`` read concern: Read operations return the instance's most recent data 
+  without guaranteeing that the data has been written to a majority of the replica
+  set members
+- ``UNACKNOWLEDGED`` write concern: Does not request acknowledgment of the write operation
+
+.. literalinclude:: /includes/read-write-pref.scala
+    :language: scala
+    :dedent:
+    :start-after: start-collection-settings
+    :end-before: end-collection-settings
+
+.. _scala-read-write-advanced:
+
+Advanced Read Configurations 
+----------------------------
+
+This section shows how to further customize your read operation
+settings in the following ways:
+
+- :ref:`Apply a tag set <scala-tag-sets>`
+- :ref:`Specify a local threshold <scala-local-threshold>`
+
+.. _scala-tag-sets:
+
+Tag Sets
+~~~~~~~~
+
+The {+mdb-server+} allows you to apply key-value :manual:`tags
+</core/read-preference-tags/>` to replica set members
+according to any criteria you choose. You can then use those
+tags to target one or more members for a read operation.
+
+By default, the {+driver-short+} ignores tags when choosing a member
+to read from. To instruct the {+driver-short+} to prefer certain tags,
+pass them as a list to your read preference setter method.
+
+Suppose you are connected to a replica set that contains members hosted
+at multiple data centers across the United States. This code example sets
+the read preference of ``test_database`` to a tag set that instructs the database
+to prefer reads from secondary replica set members in the following order:
+
+1. Members from the New York data center (``Tag("dc", "ny")``)
+#. Members from the San Francisco data center (``Tag("dc", "sf")``)
+#. Any secondary members (``TagSet()``)
```

Comment:
S: what if you moved this up after the first sentence to make it more linear?

1. suppose you have this replica set
2. the members are tagged like this
3. here's how you'd specify the read order:

Reply:
Okay I think I reordered similar to what you had in mind